File: CS/MyDateEdit/RepositoryItemMyDateEdit.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.Utils.Win;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Popup;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;

namespace WindowsFormsApplication2
{
    [UserRepositoryItem("Register")]
    public class RepositoryItemMyDateEdit : RepositoryItemDateEdit
    {
        static RepositoryItemMyDateEdit()
        {
            Register();
        }
        public RepositoryItemMyDateEdit() { }

        private bool _ShowYearArrows;
        internal const string EditorName = "MyDateEdit";

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(MyDateEdit),
                typeof(RepositoryItemMyDateEdit), typeof(DevExpress.XtraEditors.ViewInfo.DateEditViewInfo),
                new DevExpress.XtraEditors.Drawing.ButtonEditPainter(), true));
        }
        public override string EditorTypeName
        {
            get { return EditorName; }
        }

        public bool ShowYearArrows
        {
            get { return _ShowYearArrows; }
            set
            {
                if (_ShowYearArrows != value)
                {
                    _ShowYearArrows = value;
                    OnPropertiesChanged();
                }
            }
        }

        public override void Assign(RepositoryItem item)
        {
            BeginUpdate();
            try
            {
                base.Assign(item);
                RepositoryItemMyDateEdit source = item as RepositoryItemMyDateEdit;
                if (source == null) return;
                ShowYearArrows = source.ShowYearArrows;
            }
            finally
            {
                EndUpdate();
            }
        }
    }
}

File: VB/MyDateEdit/RepositoryItemMyDateEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Reflection
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.Utils.Win
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Popup
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository

Namespace WindowsFormsApplication2
	<UserRepositoryItem("Register")> _
	Public Class RepositoryItemMyDateEdit
		Inherits RepositoryItemDateEdit
		Shared Sub New()
			Register()
		End Sub
		Public Sub New()
		End Sub

		Private _ShowYearArrows As Boolean
		Friend Const EditorName As String = "MyDateEdit"

		Public Shared Sub Register()
			EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(MyDateEdit), GetType(RepositoryItemMyDateEdit), GetType(DevExpress.XtraEditors.ViewInfo.DateEditViewInfo), New DevExpress.XtraEditors.Drawing.ButtonEditPainter(), True))
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property

		Public Property ShowYearArrows() As Boolean
			Get
				Return _ShowYearArrows
			End Get
			Set(ByVal value As Boolean)
				If _ShowYearArrows <> value Then
					_ShowYearArrows = value
					OnPropertiesChanged()
				End If
			End Set
		End Property

		Public Overrides Sub Assign(ByVal item As RepositoryItem)
			BeginUpdate()
			Try
				MyBase.Assign(item)
				Dim source As RepositoryItemMyDateEdit = TryCast(item, RepositoryItemMyDateEdit)
				If source Is Nothing Then
					Return
				End If
				ShowYearArrows = source.ShowYearArrows
			Finally
				EndUpdate()
			End Try
		End Sub
	End Class
End Namespace
